### DISCUSSION TITLE
V3 Form Builder: can't use value() in TextInput

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9111] [Help] [answered]
I'm using a filament plugin to create pages what I want to achieve is to add the content in **Placeholder** into **TextInput**? I tried to use `value()` but it's not exist.

**Original Code:**

```
Placeholder::make('url')
->label(__('URL'))
->visible(fn (?PageContract $record) => config('demo.enabled') && filled($record))
->content(fn (?PageContract $record) => Demo::getPageUrlFromId($record?->id)),
```

**My Code:**

```
TextInput::make('url')
->label(__('URL'))
->visible(fn (?PageContract $record) => config('demo.enabled') && filled($record))
->value(fn (?PageContract $record) => config('app.url') . Demo::getPageUrlFromId($record?->id))
->readonly(),
```

**Comment by leandrocfe** [comment]
You can use ->getStateUsing() method

**Reply by devhoussam1998** [reply]
getStateUsing does not exist

**Reply by leandrocfe** [reply, marked as answer]
Sorry, `formatStateUsing`
`->formatStateUsing(fn (?PageContract $record) => config('app.url') . Demo::getPageUrlFromId($record?->id))`

**Reply by devhoussam1998** [reply]
thank you so mush @leandrocfe
